### PR TITLE
Add `validateField()` and use it in `isFieldValid()`

### LINKED ISSUE #28: isFieldValid unexpected behaviours
Hi,

I've got a use case for needing to validate a single field before sending some info to a rest api when clicking a button. It seems like isFieldValid should do this. However I'm having some issues with the way it works. If I provide the true argument to pre-validate the form it will then validate all other inputs on the form and show their error messages. This feels like a bad user experience as the user may not have got round to entering those fields yet.

Another issue is when you supply the prevalidate false argument then isFieldValid is assuming that some user interaction has happened and that there is an error message in the element cache for the form. I've made a fiddle that hopefully illustrates the issue.

https://jsfiddle.net/po75dqef/1/

### LINKED ISSUE #76: isFieldValid not working as expected if prevalidate is false
Hi there,

First of all, thank you for taking on the maintenance role of the library after Ryan. I've just started using the library so if I'm doing something wrong, please point me in the right direction.

Otherwise I've created a [codepen repo](https://codepen.io/Maverik/pen/vYvMavr) with the problem and this is essentially the same problem as #28 I believe.

When I try to validate a specific field **with prevalidate set to false** in order to trigger an API call, the callback is never triggered and the result of isFieldValid appears to lag behind by one call.

i.e. if the last 7 results were false:
- you'll get true on first call (this is coming from default state of control while its in pristine state)
- you'll get false up-to the 8th call (1+7)
- true on the 9th call (1+8 even though the field was valid on the 8th call).
- lets say now the input was actually invalid on the 9th call, it'll take the 10th call to actually get that result.
 
I hope you can see now what I mean by "lagging behind by one call".

The behaviour only happens with pre-validate set to false as far as I can tell and behaves as expected when working with the entire form. Of course, that leaves a pretty bad impression when the entire form says its required, even though the user hasn't even gotten to rest of the form yet.

### PR DESCRIPTION
**Pull request by dahlbyk**
- Fixes #76
- Actually fixes #28

Expected `isFieldValid()` behavior is to check a single field, not the whole form. This is _technically_ a breaking change, but the workaround is easy: call `validateForm()` (or `isValid()`) if you want to validate the whole form.

- `isValid()` calls `validateForm()` unless `prevalidate: false`
- `isFieldValid()` calls `validateField()` unless `prevalidate: false`

In both cases I tried to document that the synchronous return value isn't necessarily accurate if there's async validation.

Would welcome a demo from someone showing off how this can be used.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -956,36 +973,40 @@ export class ValidationService {
...
+                        this.logger.log('Validation error', error);
+                    });
+            }, this.debounce);
         };
 
-        let isDropdown = input.tagName.toLowerCase() === 'select';
         let validateEvent = input.dataset.valEvent;
-        if (isDropdown) {
-            input.addEventListener('change', cb);
-        } else if (validateEvent) {
```
Comment (dahlbyk) [inline src/index.ts:974]: While I'm in the area I noticed the `data-val-event` override is being ignored for `<select>`. The attribute now takes precedence over the default `change` event.

**Review by haacked — APPROVED** [review]
:shipit:

**Comment by haacked** [comment]
This is live on NPM! https://www.npmjs.com/package/aspnet-client-validation

**Comment by Maverik** [comment]
This may need a separate issue but I believe the change from this isn't reflected on README documentation yet and will likely cause some confusion around the pre-validation when used in updated release. Also it would be worth mentioning debounce functionality in that area for future users.

Thank you again for this work 🙌

**Pull request merged**